internal/http/docker.go

```diff
@@ -0,0 +1,64 @@
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+type DockerRegistry struct {
+	HTTPClient *retryablehttp.Client
+	URL        string
+	Username   string
+	AccessKey  string
+}
+
+type AuthToken struct {
+	ExpiresAt time.Time `json:"expires_at"`
+	Username  string    `json:"username"`
+	Password  string    `json:"password"`
+}
+
+func NewDockerRegistry(url, username, accessKey string, timeout time.Duration) DockerRegistry {
+	return DockerRegistry{
+		HTTPClient: NewRetryableClient(timeout),
+		URL:        url,
+		Username:   username,
+		AccessKey:  accessKey,
+	}
+}
+
+func (c *DockerRegistry) Login(ctx context.Context, repo string) (AuthToken, error) {
+	url := fmt.Sprintf("%s/v1alpha1/hosted/container-registry/%s/authorization-token", c.URL, repo)
```

Comment:
It does require the `repo` here. I've tried without it and got permission denied.

```
{"errorDetail":{"message":"denied: Permission \"artifactregistry.repositories.uploadArtifacts\" denied on resource \"projects/sauce-hto-p-jy6b/locations/us-west4/repositories/sauce-devx-team-sauce\" (or it may not exist)"},"error":"denied: Permission \"artifactregistry.repositories.uploadArtifacts\" denied on resource \"projects/sauce-hto-p-jy6b/locations/us-west4/repositories/sauce-devx-team-sauce\" (or it may not exist)"}
```